qiskit_qec/operators/base_xp_pauli.py

```diff
@@ -53,7 +53,7 @@ def __init__(
         self,
         matrix: Union[np.ndarray, None] = None,
         phase_exp: Union[None, np.ndarray, np.integer] = None,
-        precision: int = None,
+        precision: Union[int, np.ndarray] = None,
```

Comment:
The precision for each Xp operator in the base should be the same as we are packing them into a single matrix and will be Z/NZ operations on them. Was there some other reason for having an array of precisions? Yes we could have an list of XP operators of different precisions but this is actually unlikely. We could have an input method that allows you to input operators of different precisions in but we would have to lift these operators to a common precision.

Reply:
@dsvandet Yes, I will revert this. The reason to introduce different precisions was because how the function `XPSetN` (now, `rescale_precision` in `BaseXPPauli`) works in Mark's XPF package [(see here)](https://github.com/m-webster/XPFpackage/blob/1a58c31d282bb2ca97591a0433a479559aef65c3/XPAlgebra.py#L124) - in a list of operators, it modifies operator `A[i]` with precision `N[i]` to new precision `P[i]`. However, in our last meeting, Mark (@m-webster) confirmed that this functionality has no significant use. So I agree with you.